src/utils/produceScoutReport.tsx

```diff
@@ -1,99 +0,0 @@
-import { Unit } from "../types";
-
-export function ScoutReport(
-  friendlyArmy: Unit[],
-  friendlyPowerLevel: number,
-  enemyArmy: Unit[],
-  enemyPowerLevel: number,
-  enemyUnitTypes: number
-) {
-  const enemyIsTwiceAsLarge = enemyArmy.length > 2 * friendlyArmy.length;
-  const enemyIsApproxSameSize =
-    Math.abs(enemyArmy.length - friendlyArmy.length) <
-    0.1 * friendlyArmy.length;
-  const enemyIsHalfSize = enemyArmy.length < 0.5 * friendlyArmy.length;
-  const enemyIsTwiceAsPowerful = enemyPowerLevel > friendlyPowerLevel * 2;
-  const enemyIsApproxSamePower =
-    Math.abs(enemyPowerLevel - friendlyPowerLevel) < 0.1 * friendlyPowerLevel;
-  const enemyIsHalfPowerful = enemyPowerLevel < 0.5 * friendlyPowerLevel;
```

Comment:
I think what you can do is just do `enemyArmy.length / friendlyArmy.length` to get a percentage. Then instead of doing the percent checking here you can do it in the parent code. This should allow you to add as many ranges as you like without having to explicitly do so here. Also since the ratio is just a number and not a list of possible booleans, its just more flexible.

Also don't return any UI (HTML elements) from utility code. They should be separated.